### PR TITLE
Add annotation to activate UI validation over external cluster metadata

### PR DESCRIPTION
**Pull request by bipuladh**
Adds an annotation in the CSV whose value is an object. It is required by the Console to perform validation on the key/value pairs required to connect to the external cluster. The validation is performed on the JSON file provided by the user(which in turn is generated by a script).

### CODE REVIEW

File: tools/csv-merger/csv-merger.go [hunk 1]
```diff
@@ -468,12 +468,12 @@ func generateUnifiedCSV() *csvv1.ClusterServiceVersion {
 			ocsCSV.Spec.CustomResourceDefinitions.Owned[i].DisplayName = "Storage Cluster"
 			ocsCSV.Spec.CustomResourceDefinitions.Owned[i].Description = "Storage Cluster represents a OpenShift Container Storage Cluster including Ceph Cluster, NooBaa and all the storage and compute resources required."
 			ocsCSV.Spec.CustomResourceDefinitions.Owned[i].Resources = []csvv1.APIResourceReference{
-				csvv1.APIResourceReference{
```
Comment (bipuladh) [inline tools/csv-merger/csv-merger.go:471]: @umangachapagain the script did this. IDK why.. :/

Comment (umangachapagain) [reply tools/csv-merger/csv-merger.go:471]: Script does not change this file, it only build this.

Comment (umangachapagain) [reply tools/csv-merger/csv-merger.go:471]: This is probably due to new golang changes.
Can you separate this from the actual annotation work? make 2 different commits.

Comment (bipuladh) [reply tools/csv-merger/csv-merger.go:471]: ah okay. Understood. thanks.

Comment (bipuladh) [reply tools/csv-merger/csv-merger.go:471]: Do you mind if I revert these changes and make a single commit?

Comment (umangachapagain) [reply tools/csv-merger/csv-merger.go:471]: Sure.

File: tools/csv-merger/csv-merger.go [hunk 2]
```diff
@@ -623,6 +623,7 @@ The NooBaa operator deploys and manages the [NooBaa][2] Multi-Cloud Gateway on O
 	// This can be used to enable/disable console support for any supported feature
 	// Example: "features.ocs.openshift.io/enabled": `["external", "foo1", "foo2", ...]`
 	ocsCSV.Annotations["features.ocs.openshift.io/enabled"] = `["external"]`
+	ocsCSV.Annotations["external.ocs.openshift.io/validation"] = `{"encodedKeys":["rook-ceph-operator-creds", "rook-csi-rbd-node", "rook-csi-rbd-provisioner", "rook-csi-cephfs-node", rook-csi-cephfs-provisioner"], "plainKeys": ["rook-ceph-mon-endpoints", "rook-ceph-retain-bucket", "rook-ceph-mon"]}`
```
Comment (jarrpa) [inline tools/csv-merger/csv-merger.go:626]: I would rather just have a map of `kind` to a list of names.
```suggestion
	ocsCSV.Annotations["external.ocs.openshift.io/validation"] = `{"secrets":["rook-ceph-operator-creds", "rook-csi-rbd-node", "rook-csi-rbd-provisioner", "rook-csi-cephfs-node", rook-csi-cephfs-provisioner"], "configMaps": ["rook-ceph-mon-endpoints", "rook-ceph-mon"], "storageClasses":["rook-ceph-retain-bucket"]}`
```
I would also change the annotation key to `external.features.ocs.openshift.io/jsonValidation`.

In addition, this needs a comment in the code explaining what it does and who uses it.

**Comment by bipuladh** [comment]
/hold 
Need to adjust the validation

**Comment by bipuladh** [comment]
/hold cancel
No changes required in CSV

File: tools/csv-merger/csv-merger.go [hunk 3]
```diff
@@ -623,6 +623,9 @@ The NooBaa operator deploys and manages the [NooBaa][2] Multi-Cloud Gateway on O
 	// This can be used to enable/disable console support for any supported feature
 	// Example: "features.ocs.openshift.io/enabled": `["external", "foo1", "foo2", ...]`
 	ocsCSV.Annotations["features.ocs.openshift.io/enabled"] = `["external"]`
+	// Used by UI to validate user uploaded metdata
+	// Metadata is used to connect to an external cluster
+	ocsCSV.Annotations["external.ocs.openshift.io/validation"] = `{"secrets":["rook-ceph-operator-creds", "rook-csi-rbd-node", "rook-csi-rbd-provisioner", "rook-csi-cephfs-node", rook-csi-cephfs-provisioner"], "configMaps": ["rook-ceph-mon-endpoints", "rook-ceph-mon"], "storageClasses": ["rook-ceph-retain-bucket"]}`
```
Comment (jarrpa) [inline tools/csv-merger/csv-merger.go:628]: This is still a bad URI, as it does not follow the pattern established by the line above. Please change to `external.features.ocs.openshift.io/validation`.

**Comment by bipuladh** [comment]
/hold
@umangachapagain @jarrpa please remove the hold once the CI is fixed.
No point in exhausting our resources.

**Pull request merged**